### ISSUE TITLE
SA1629 bug

### ISSUE THREAD

**Issue by vweijsters** [DotNetAnalyzers/StyleCopAnalyzers#2680]
The following code:

```c#
        /// <summary>
        /// The ID for diagnostics produced by the <see cref="SA1127GenericTypeConstraintsMustBeOnOwnLine"/>
        /// </summary>
        public const string DiagnosticId = "SA1127";
```
will flag a missing period just after the `the` word, where it should be after the `<see .../>` tag.

**Comment by sharwell** [comment]
📝 Note that the violation can be manually fixed by adding the period after the `<see>` tag. This bug only affects the reporting location.

**Comment by vweijsters** [comment]
It also affects the code fix, as due to the wrong reporting location the period is inserted in the wrong spot.

**Comment by IngbertPalm** [comment]
SA1629 should also accept other punctuation characters like '!' or '?' as valid line endings.
